rocAL/rocAL/source/pipeline/ring_buffer.cpp

```diff
@@ -351,9 +419,33 @@ void RingBuffer::increment_write_ptr()
     _wait_for_load.notify_all();
 }
 
-void RingBuffer::set_meta_data( ImageNameBatch names, pMetaDataBatch meta_data)
+void RingBuffer::set_meta_data(ImageNameBatch names, pMetaDataBatch meta_data, bool is_segmentation)
 {
-    _last_image_meta_data = std::move(std::make_pair(std::move(names), meta_data));
+    if(meta_data == nullptr)
+        _last_image_meta_data = std::move(std::make_pair(std::move(names), pMetaDataBatch()));
+    else
+    {
+        _last_image_meta_data = std::move(std::make_pair(std::move(names), meta_data));
+        if(!_box_encoder_gpu)
```

Comment:
Where is this set?

Reply:
@fiona-gladwin Please check this

Reply:
Not used anywhere and not set any where. So, removing it

Reply:
Should not be removed @fiona-gladwin